File: src/main/java/net/ccbluex/liquidbounce/utils/animations/impl/DecelerateAnimation.java
package net.ccbluex.liquidbounce.utils.animations.impl;


import net.ccbluex.liquidbounce.utils.animations.Animation;

public class DecelerateAnimation extends Animation {

    public DecelerateAnimation(int ms, double endPoint) {
        super(ms, endPoint);
    }

    protected double getEquation(double x) {
        return 1 - ((x - 1) * (x - 1));
    }
}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/nlclickgui/animations/impl/DecelerateAnimation.kt
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.impl

import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Animation
import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Direction

class DecelerateAnimation @JvmOverloads constructor(
    ms: Int,
    endPoint: Double,
    direction: Direction = Direction.FORWARDS
) : Animation(ms, endPoint, direction) {

    override fun getEquation(x: Double): Double {
        val x1 = x / duration
        return 1 - (x1 - 1) * (x1 - 1)
    }
}
